### ISSUE TITLE
Key not synced properly

### ISSUE THREAD

**Issue by Coecoenut** [LibreSignal/LibreSignal#13]
Hi,
I have an issue with sending messages to a Signal user. I am running LibreSignal 3.13.0, the other person is using Signal in the most recent version from Google Play.
While I can receive encrypted messages from this person, I cannot answer them. Instead of sending the message via Signal, a simple SMS is sent, and I can see the message "Recipient has no identity key".
I also cannot view the key of this user, while the other person can view my key.
The log of my device can be found here:
https://gist.github.com/Coecoenut/2c70637781dae9f490f4#file-libresignal-log

I checked the behaviour from a second LibreSignal user who can communicate encrypted with the other person without any issues. They can also view their keys.

**Comment by mimi89999** [comment]
You and your contact should:
1. Open the conversation
2. Open the menu: 3dots in the top right corner
3. Select "reset secure conversation"

**Comment by Coecoenut** [comment]
I don't have this option on my device. The chat is treated as if the other person had no Signal at all.
I will contact the other person and will let you know if this solved the issue.

**Comment by Coecoenut** [comment]
The other person reseted the conversation several times (in some cases I got informed that the encryption was restartet, but most of the time it failed and I was not informed), but I still do not have the key.

When the reset of the conversation fails, the other person is informed by the message "nicht zugestellt" (not transmitted). Details on this message inform about events that require the readers attention, that the transmission to the respective person failed, and there is a button with the label "Erneut senden" (send again).

The person switched now from Signal to LibreSignal. During the first reset of the conversation after the switch, I was informed about a message including the key (which I could check). But for some reason the key was not saved.
So the situation is exactly as before, except of the fact that both dialog partners are using LibreSignal now.

**Comment by mimi89999** [comment]
Did you try reinstalling the app?

**Comment by Coecoenut** [comment]
Yes. Reinstalling did not change the situation.

**Comment by mimi89999** [comment]
How is the situation with other contacts?

**Comment by Coecoenut** [comment]
I did not encounter any issues with other contacts.

**Comment by mimi89999** [comment]
And the other person with his contacts?

**Comment by Coecoenut** [comment]
Also no issues - although there is only a limited number of contacts on the other side, so I am not sure how significant this information is.
Are there any data on my device which could survive a reinstallation of LibreSignal, which I could remove manually?

**Comment by mimi89999** [comment]
Do you have Gapps on your device? Does the other side have Gapps on his device?

**Comment by Coecoenut** [comment]
Both sides have Gapps.

**Comment by mimi89999** [comment]
> Are there any data on my device which could survive a reinstallation of LibreSignal, which I could remove manually?

On your phone I don't think so. You can try to unregister from OWS servers: https://whispersystems.org/textsecure/unregister/

**Comment by mimi89999** [comment]
Did you install the app from Eutopia experimental repo?

**Comment by Coecoenut** [comment]
I just removed LibreSignal and unregistered from OWS servers, then reinstalled LS.
After starting a conversation with the respective contact, I only get the option to send SMS, while for my other (Libre-/)Signal contacts I can start an encrypted conversation.

> Did you install the app from Eutopia experimental repo?

If https://eutopia.cz/fdroid/repo is "experimental", yes.

**Comment by mimi89999** [comment]
I think the problem might be on the other side...

**Comment by mimi89999** [comment]
The Eutopia experimental repo is: https://eutopia.cz/experimental/fdroid/

You can read more on: https://fdroid.eutopia.cz/

**Comment by Coecoenut** [comment]
Ok, I just added the repository and updated LibreSignal. I got informed that my Google Services are broken, and had to re-register LibreSignal.
Situation is like before, I can communicate with my other contacts with encryption, but I can only send SMS to this single contact.
I will ask the person to add the experimental repo to F-Droid, update LS and check if the issue still exists. I will let you know about ASAP - thank you for your effort btw!

**Comment by mimi89999** [comment]
Maybe it's a stupid question...
Your device appears to be dual SIM. Is your contact contacting you on the number you are registered with LibreSignal? Are you contacting the other person on the phone number he/she is registered on LibreSignal?

**Comment by Coecoenut** [comment]
Yes, the device is dual SIM, but I am using only one SIM slot currently - there is no second SIM card (yet).

**Comment by mimi89999** [comment]
Could I get the log from the other device?

**Comment by mimi89999** [comment]
Is the other device dual SIM?

**Comment by Coecoenut** [comment]
Nope, the other device is single SIM.
I will ask the person for the log. Do you want the log before updating to the version in experimental or after?

**Comment by mimi89999** [comment]
Without updating, just after trying to send you a message.

**Comment by mimi89999** [comment]
Also: The version in the main Eutopia repo is built from whispersystems Signal-Android repo.

**Comment by mimi89999** [comment]
@Coecoenut There is a new version of LibreSignal in the Eutopia experimental F-Droid repo

**Comment by Coecoenut** [comment]
Thank you. I updated to the new version, but there are no changes about the error from my side.
I will let you know when I have the log from the other side. Should happen this weekend.

> Also: The version in the main Eutopia repo is built from whispersystems Signal-Android repo.

I knew this, although I thought that the version in the repo was already using WebSocket. But thank you for making this clear.
My report in the GitHub of Signal was closed by Moxie immediately after I mentioned "Libre" in the text.
Oh if this helps, there was also a similar report including an iOS device:
https://github.com/WhisperSystems/Signal-iOS/issues/1048

**Comment by mimi89999** [comment]
If you want, you can check if you can see my key. I'm registered with the phone number "+48 532 465 056"
I will delete this comment not to receive SPAM... I hope you get notifications by email...

**Comment by Coecoenut** [comment]
I checked your number. I can see you and your key from my device.

**Comment by mimi89999** [comment]
I think the problem is on the other device.

**Comment by mimi89999** [comment]
Any news?

**Comment by Coecoenut** [comment]
Unfortunately not. I could not contact the person yet.

**Comment by mimi89999** [comment]
Do you know what Device/ROM the other person is using?

**Comment by Coecoenut** [comment]
I think it's a LG Optimus 4X HD, using Cyanogenmod 11

**Comment by Coecoenut** [comment]
So here is the log of the other device:
https://gist.github.com/Coecoenut/c1af2444ad09bcb16e73
The log was created after trying to send a message to my device, which failed.

**Comment by mimi89999** [comment]
@JavaJens Could you please look at this?

**Comment by mimi89999** [comment]
`W/PushTextSendJob( 4385): org.whispersystems.textsecure.api.crypto.UntrustedIdentityException: Untrusted identity key!`
Did the other person accept your key?

**Comment by Coecoenut** [comment]
As far as I know: Yes.
Otherwise it wouldn't be possible to send encrypted messages to my device, right?
But I will ask to make sure that the information is correct.

**Comment by mimi89999** [comment]
Any news?

**Comment by Coecoenut** [comment]
Yes and no. I didn't get an answer yet (and unfortunately this could take several weeks now), but I checked this contact again. It doesn't say anymore that my contact is using (Libre)Signal, but I can only send a regular SMS or invite to Signal.
But the person is still in the list of available Signal contacts.

**Comment by mimi89999** [comment]
If the problem isn't resolved, could you please check it in Signal from Play Store? If it's also present there, please open an issue upstream...
